src/std/db/postgresql-driver.ss

```diff
@@ -95,6 +135,24 @@
       (!!postgresql.reset driver token))
     (error "Bad argument; illegal query token" token)))
 
+(def (postgresql-current-notice-handler conn)
+  (alet (driver (get-driver conn))
+    (!!postgresql.current-notice-handler driver #f)))
+
+
+(def (postgresql-current-notice-handler-set! conn handler)
+  (alet (driver (get-driver conn))
+    (!!postgresql.current-notice-handler driver handler)))
+(def (with-postgresql-notice-handler conn handler thunk)
+  (let (pren (postgresql-current-notice-handler conn))
+    (try
+     (set! (postgresql-current-notice-handler conn)
+       (lambda args (apply handler args) (apply pren args)))
```

Comment:
Yeah, that was the intention. Because we allow query results and therefore notices to appear "anywhere" and because of the new -query struct even nest them I thought that the "next-method" apply was a good idea. Beyond that I like that default warnf alongside the notice gathering. 

We could always duplicate it, but I prefer this way. Thoughts?

Reply:
fair enough.